### PR TITLE
Adding LoadUserConsent and validation logic

### PR DESCRIPTION
**Pull request by oscarraigscmspain**
## Description

- When boros loadUserConsent is called:
  - Handle EventStatus "tcloaded" (related with addEventListener)
  - If consent is valid, set LOADED CmpStatus
  - Retrieve getVendorList and :
     - if vendorList version are equal return valid consent
     - if vendorList versions are different:
         - If all consents were accepted, return valid consent, and add new vendors
         - If all consents were rejecte, return valid consent, and reject new vendors
         - othewise return no valid consent
## Solves ticket/s

https://jira.scmspain.com/browse/PSP-3201
## Expected behavior

- Return consent with valid attribute set
## Review steps

- This is a middle stage
## Further considerations

- In further iterations maybe we should loadConsent at some part of the initialization
## Memetized description

mandatory

### CODE REVIEW

File: src/main/domain/vendorlist/VendorListHelper.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    return true
+  }
+
+  setAllVendorsTo({vendorList, valueToSet}) {
+    const newVendor = {
+      consents: {},
+      legitimateInterests: {}
+    }
+    Object.entries(vendorList).forEach(([key, value]) => {
```
Comment (maryammdot) [inline src/main/domain/vendorlist/VendorListHelper.js:20]: If you are not checking the value of vendorList object here you can use Object.keys

File: src/main/domain/vendorlist/VendorListHelper.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return newVendor
+  }
+
+  mergeVendors({newVendorList, oldVendors}) {
+    const mergedVendorList = {
+      consents: {},
+      legitimateInterests: {}
+    }
+
+    Object.entries(newVendorList).forEach(([key, value]) => {
```
Comment (maryammdot) [inline src/main/domain/vendorlist/VendorListHelper.js:33]: Same as before with Object Keys. Also, why do you need to parseInt the key?

File: src/main/domain/consent/LoadConsentService.js [hunk 3]
```diff
@@ -2,25 +2,118 @@ import {inject} from '../../core/ioc/ioc'
...
+    existingConsent.vendor = this._vendorListHelper.mergeVendors({
+      newVendorList: newVendorList.vendors,
+      oldVendors: existingConsent.vendor
+    })
+    return this._consentFactory.createConsent(existingConsent)
+  }
+
+  async _createValidConsentWithTheSavedOne() {
+    const validEncodedConsent = await this._consentRepository.loadUserConsent()
+    const validConsent = this._consentDecoderService.decode({
```
Comment (maryammdot) [inline src/main/domain/consent/LoadConsentService.js:62]: Is it necessary to get the recently stored consent (in _consentHaveConsentsAndLIForAllVendorsWithTheSameValue method) from the consent repository? Or we could get the one that that method is working with?

File: src/main/domain/vendorlist/VendorListHelper.js (hunk 1 above)
Comment (oscarraigscmspain) [reply src/main/domain/vendorlist/VendorListHelper.js:20]: done. Thanks!

File: src/main/domain/vendorlist/VendorListHelper.js (hunk 2 above)
Comment (oscarraigscmspain) [reply src/main/domain/vendorlist/VendorListHelper.js]: Ok, using Object keys, and Ok removing parse int. done

File: src/main/domain/consent/LoadConsentService.js (hunk 3 above)
Comment (oscarraigscmspain) [reply src/main/domain/consent/LoadConsentService.js:62]: I have preferred return, the saved one. For performance reasons it is better return the one that method is working with, but performance issues I think is better to attach later.

File: src/main/application/services/vendorconsent/LoadUserConsentUseCase.js [hunk 4]
```diff
@@ -1,15 +1,28 @@
...
   }
 
   async execute() {
-    // TODO dispatch required IAB event
     const consent = await this._loadConsentService.loadConsent()
-    return consent.toJSON()
+    const consentDto = consent.toJSON()
+    if (consentDto.valid) {
+      this._status.cmpStatus = Status.CMPSTATUS_LOADED
+      this._eventStatusService.updateTCLoaded()
```
Comment (alextremp) [inline src/main/application/services/vendorconsent/LoadUserConsentUseCase.js:23]: so every time somebody requests the User Consent, an event will be raised?

Comment (oscarraigscmspain) [reply src/main/application/services/vendorconsent/LoadUserConsentUseCase.js:23]: We can discuss, and add a parameter notify, default value = false 
notify && this._eventStatusService.updateTCLoaded()
Then in TcfInitializer 
const borosTcf() = new BorosTcf()
borosTcf().loadUserConsent.execute({notify:true})
return borosTcf. 
But I am thinking to do all this in a separate PR. Moving event to initialization will break a lot of tests, and I prefer doing it in a separate PR.

**Pull request merged**